python/spacewalk/satellite_tools/repo_plugins/yum_src.py

```diff
@@ -683,11 +684,20 @@ def _prep_zypp_repo_url(self, url, uln_repo):
         if self.sslclientkey:
             query_params['ssl_clientkey'] = self.sslclientkey
         new_query = unquote(urlencode(query_params, doseq=True))
-        if self.authtoken or uln_repo:
-            ret_url = "{0}&{1}".format(url, new_query)
-        else:
-            ret_url = "{0}?{1}".format(url, new_query) if new_query else url
-        return ret_url
+        # urlparse cannot handle uln urls, so we need to keep this check
+        if uln_repo:
+            return "{0}&{1}".format(url, new_query)
+        parsed_url = urlparse(url)
+        query = parsed_url.query
+        final_query = "&".join(filter(None, [query, new_query]))
```

Comment:
The names `query`, `new_query` and `final_query` are a bit confusing, it's unclear what the purpose of each of them is. Naming is hard, but I think we can come up with better fitting names here.

Reply:
Well, query is the query that already is in the url. The new query is the one we add in this method and is composed of the proxy and ssl certificates. I can see that we should change final_query, but the other two are just that. A query and a new query we add to the url.

Reply:
Sounds like you can just reuse `new_query` here.

Reply:
I think if I rename them into `existing_query`, `new_query` and `combined_query` then it should be clear